app/helpers/page_helper.rb

```diff
@@ -1,2 +1,13 @@
 module PageHelper
+  def practices_to_display_in_info_window(facility_id, practice_list, names, taglines, short_names, slugs)
+    dh = DiffusionHistory.where(va_facility_id: facility_id)
+    dh.each do |dh|
+      if practice_list.practices.include?(dh.practice_id.to_s)
+        names.push dh.practice.name
+        short_names.push dh.practice.short_name
+        taglines.push dh.practice.tagline
+        slugs.push dh.practice.slug
```

Comment:
1. I think we should rename this method as well. Perhaps something like `get_practices_by_diffusion_histories`.
2. Instead of creating 4 different arrays, we can use the `:get_with_practice` scope method in `models/diffusion_history.rb`. Once we have the practice, we can use .slice to create a hash that contains the keys and value for each column we need. Ex: 
```
{"name"=>"Some practice name", "short_name"=>"Some short name"}
```
3. We can then push each one of these into a single array and then loop through them.
4. We should write a unit test to ensure it works properly.

Reply:
This sounds like a good suggestion for reducing database calls.

Reply:
Done.